File: 1706-where-will-the-ball-fall/1706-where-will-the-ball-fall.py
class Solution:
    def findBall(self, grid: List[List[int]]) -> List[int]:
        m = len(grid)
        n = len(grid[0])
        res = [i for i in range(n)]
        
        for i in grid:
            for j, val in enumerate(res):
                if val!=(-1):
                    if (val + i[val])%(n+1) == n:
                        res[j] = -1
                    elif i[val]!=i[val+i[val]]:
                        res[j] = -1
                    else:
                        res[j] = val + i[val]
        
        return res

File: 1706-where-will-the-ball-fall/1706-where-will-the-ball-fall.cpp
class Solution {
public:
    vector<int> findBall(vector<vector<int>>& grid) {
        int m,n,i,row,col;
        m = grid.size(), n = grid[0].size();
        
        vector<int> solution(n);
        for(i=0 ; i<n ; i++){            
            bool ballStuck = false;            
            row = 0, col = i;
            for( ; row<m ; row++){
                if(grid[row][col] == 1){
                    if(col <= (n-2) && grid[row][col+1] == 1){
                        col++;
                    }
                    else{
                        ballStuck = true;
                        break;
                    }
                }
                else{
                    if(col >= 1 && grid[row][col-1] == -1){
                        col--;
                    }
                    else{
                        ballStuck = true;
                        break;
                    }
                }
            }
            
            solution[i] = (ballStuck)? -1 : col;            
        }
        
        return solution;
    }
};